cabal-install/Distribution/Client/CmdRun.hs

```diff
@@ -309,11 +311,11 @@ selectPackageTargets targetSelector targets
   where
     (targetsExesBuildable,
      targetsExesBuildable') = selectBuildableTargets'
-                            . filterTargetsKind ExeKind
+                            . (\ts -> concatMap (`filterTargetsKind` ts) [ExeKind, TestKind, BenchKind])
```

Comment:
`ExeKind` was first argument, now it's (`ts`) is second. Is `filterTragetsKind

Reply:
What do you mean? The behavior is the same (note the backticks).

Uuh anyway this whole patch is not in a good shape right now. No need to review it until I polish it

Reply:
Ah I got confused. Maybe `(\kind -> filterTargetsKind kind ts)` would be less mind-twisting, or  adding `filterTargetKinds :: [ComponentKind] -> ...`